client/my-sites/plugins/education-footer/index.tsx

```diff
@@ -155,15 +156,27 @@ const EducationFooter = () => {
 	const { __ } = useI18n();
 	const localizeUrl = useLocalizeUrl();
 	const dispatch = useDispatch();
+	const { openArticleInHelpCenter } = useOpenArticleInHelpCenter();
+	const isLoggedIn = useSelector( isUserLoggedIn );
 
 	const onClickLinkCard = useCallback(
-		( content_type: string ) => {
+		( content_type: string, url: string ) => ( e: React.MouseEvent< HTMLAnchorElement > ) => {
 			dispatch(
 				recordTracksEvent( 'calypso_plugins_educational_content_click', { content_type } )
 			);
+			if ( isLoggedIn ) {
```

Comment:
Suggestion to fix the dependency array warning and avoid what the AI calls "unnecessary nested function creation on each render," or, only calling the function if logged in.

```
const onClickLinkCard = useCallback(
	(contentType: string, url: string) => {
		dispatch(
			recordTracksEvent('calypso_plugins_educational_content_click', { content_type: contentType })
		);

		if (isLoggedIn) {
			return (e: React.MouseEvent<HTMLAnchorElement>) => {
				e.preventDefault();
				openArticleInHelpCenter(url);
			};
		}

		return undefined;
	},
	[dispatch, openArticleInHelpCenter, isLoggedIn]
);
```

Reply:
Thanks, Allison. I tried applying this, but it breaks the click tracking. Can you give me more details on the "dependency array warnings"? I'm not seeing them, or I might be missing them. 😅

Reply:
This is the warning I'm seeing:

<img width="1171" alt="Screen Shot 2024-09-16 at 9 56 28 AM" src="https://github.com/user-attachments/assets/dfb11205-0a48-4dd2-bf55-a6ec66c8d476">

Ignoring the `return` suggested since it broke tracking, you can fix it by adding `isLoggedIn` to the dependency array:

```
[dispatch, openArticleInHelpCenter, isLoggedIn]
```

Reply:
Ah, I didn't have ESLint installed locally on Cursor. I see the warning now. 😅 Thanks! -- I added isLoggedIn to the dependency array.